src/Tribe/Admin/Notice/Install_Event_Tickets.php

```diff
@@ -192,7 +193,67 @@ public function should_display_notice_activate(): bool {
 		 *
 		 * @param bool $should_display True if the notice should display.
 		 */
-		return apply_filters( 'tec_events_admin_notice_event_tickets_should_display', $this->is_installed() && ! $this->is_active() && ! $this->is_install_plugin_page() );
+		return apply_filters( 
+			'tec_events_admin_notice_event_tickets_should_display', 
+			$this->is_installed() && ! $this->is_active() && ! $this->is_install_plugin_page() && $this->is_tec_related_page()
+		);
+	}
+
+	/**
+	 * Checks if the current admin page is TEC related.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool True if the current admin page is TEC related.
+	 */
+	public function is_tec_related_page(): bool {
+		// Not in the admin we don't even care.
+		if ( ! is_admin() ) {
+			return false;
+		}
+
+		global $current_screen;
+		
+		// No screen, bail.
+		if ( empty( $current_screen ) ) {
+			return false;
+		}
+		
+		// Match any TEC post type screens (events, organizers, venues).
+		$tec_post_types = [
+			\Tribe__Events__Main::POSTTYPE,
+			\Tribe__Events__Organizer::POSTTYPE,
+			\Tribe__Events__Venue::POSTTYPE
+		];
+		
+		if ( in_array( $current_screen->post_type, $tec_post_types, true ) ) {
+			return true;
+		}
+		
+		// Match any screen ID containing 'tribe_events'.
+		if ( false !== strpos( $current_screen->id, \Tribe__Events__Main::POSTTYPE ) ) {
+			return true;
+		}
+		
+		// Match any screen ID containing 'tec-'.
+		if ( false !== strpos( $current_screen->id, 'tec-' ) ) {
```

Comment:
Would this be better with `str_starts_with()`? Would we care if `tec-` is found somewhere else in the string (even if unlikely)?